### PR TITLE
better binary filtering UX + hybrid regexes + max column preview + and more

### PR DESCRIPTION
**Pull request by BurntSushi**
This PR fixes several bugs and adds a few new features that I want to get into the next release. Notably, this prepares the next release to be `ripgrep 11.0.0` and _not_ `ripgrep 0.11.0`.

### CODE REVIEW

File: complete/_rg [hunk 1]
```diff
@@ -270,6 +270,8 @@ _rg() {
     '*--ignore-file=[specify additional ignore file]:ignore file:_files'
     '(-v --invert-match)'{-v,--invert-match}'[invert matching]'
     '(-M --max-columns)'{-M+,--max-columns=}'[specify max length of lines to print]:number of bytes'
+    '--max-column-preview[show preview for long lines when limit is enabled]'
```
Comment (okdana) [inline complete/_rg]: These two should be made exclusive with each other, probably by breaking them out into a group with `+ '(max-column-preview)'` or similar

Also, i would suggest the description `show preview for long lines (with -M)` — that matches the wording for similar options, and might be a little clearer

File: complete/_rg [hunk 2]
```diff
@@ -182,6 +182,7 @@ _rg() {
     + '(pcre2)' # PCRE2 options
     {-P,--pcre2}'[enable matching with PCRE2]'
     $no'(pcre2-unicode)--no-pcre2[disable matching with PCRE2]'
+    '--pcre2-version[print the version of PCRE2 used by ripgrep, if available]'
```
Comment (okdana) [inline complete/_rg]: If i'm understanding correctly (that it has basically the same usage semantics as `--version`), this belongs under the `+ '(exclusive)'` group

File: complete/_rg [hunk 3]
```diff
@@ -44,6 +44,10 @@ _rg() {
     '(: * -)'{-h,--help}'[display help information]'
     '(: * -)'{-V,--version}'[display version information]'
 
+    + '(hybrid)' # hybrid regex options
+    '--auto-hybrid-regex[dynamically use PCRE2 if necessary]'
```
Comment (okdana) [inline complete/_rg]: Kind of a nit-pick, but describing the affected functionality differently between the enable and disable options feels a little odd (and inconsistent with other options)

**Comment by okdana** [comment]
`--max-column-preview` works really nicely, thank you for going to the trouble of implementing it (and the other stuff too, obv)

**Comment by BurntSushi** [comment]
@okdana Thanks very much for your review. :-) I think everything you pointed out should be fixed! And thanks for testing the preview functionality. I'm near certain that there will be bugs there because correctly testing the color spans was more work than I wanted to do.

**Comment by BurntSushi** [comment]
@okdana If all goes well with the max column preview, I'm contemplating turning it on by default in ripgrep 12. You were completely right about it being the "expected" behavior here. (I've put it in my .ripgreprc.)

**Comment by okdana** [comment]
Oh, actually, on reflection, do you think it should be `--max-columns-preview` (with an `s`) instead? That was the name you originally suggested in #1078, and it seems to make sense since it enables a `preview` for `--max-columns`. I'm sure you put some more thought into the name, but i thought i'd bring it up now, just in case, before it's completely set in stone

**Comment by BurntSushi** [comment]
I don't think I thought about it too much. I just felt like `--max-column-preview` sounded better in my head. Do you like `--max-columns-preview` better?

**Comment by okdana** [comment]
For what it's worth, yeah, i think so

**Comment by BurntSushi** [comment]
@okdana All righty, SGTM. Done!

**Pull request merged**